stock_scanner/hardware/sentinel.py

```diff
@@ -0,0 +1,866 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# © 2011-2015 Sylvain Garancher <sylvain.garancher@syleam.fr>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import os
+import sys
+import math
+import types
+import gettext
+import textwrap
+import traceback
+import ConfigParser
+import curses.ascii
+import unicodedata
+from datetime import datetime
+from oobjlib.connection import Connection
+from oobjlib.component import Object
+from functools import reduce
+
+# Translation configuration
+I18N_DIR = '%s/i18n/' % os.path.dirname(os.path.realpath(__file__))
+I18N_DOMAIN = 'sentinel'
+I18N_DEFAULT = 'en_US'
+
+# _ will be initialized by gettext.install but prevent pep8 issues
+_ = lambda a: a  # noqa
+
+# Default configuration
+DEFAULT_CONFIG = {
+    'host': 'localhost',
+    'user': 'admin',
+    'password': 'admin',
+    'port': '8069',
+    'test_file': None,
+}
+
+# Names of the ncurses colors
+COLOR_NAMES = {
+    'black': curses.COLOR_BLACK,
+    'blue': curses.COLOR_BLUE,
+    'cyan': curses.COLOR_CYAN,
+    'green': curses.COLOR_GREEN,
+    'magenta': curses.COLOR_MAGENTA,
+    'red': curses.COLOR_RED,
+    'white': curses.COLOR_WHITE,
+    'yellow': curses.COLOR_YELLOW,
+}
+
+# Pre-defined color pairs
+COLOR_PAIRS = {
+    'base': (1, 'white', 'blue'),
+    'info': (2, 'yellow', 'blue'),
+    'error': (3, 'yellow', 'red'),
+}
+
+
+class Sentinel(object):
+
+    """
+    Sentinel class
+    Manages scanner terminals
+    """
+
+    def __init__(self, stdscr):
+        """
+        Initialize the sentinel program
+        """
+        # Read user configuration
+        config = ConfigParser.SafeConfigParser(DEFAULT_CONFIG)
+        config.read('.oerp_sentinelrc')
```

Comment:
Maybe add a second possible name, to avoid breaking already in production instances ?